### PR TITLE
feat: enhance password strength validation

### PR DESCRIPTION
**Pull request by pratyushsingha**
[issue #29 ](https://github.com/Abbhiishek/codecommunitymusic-client/issues/29)

### CODE REVIEW

**Comment by Abbhiishek** [comment]
https://vercel.live/open-feedback/codecommunitymusic-git-fork-pratyushsingha-lo-7d2832-abbhiishek.vercel.app?via=pr-comment-visit-preview-link&passThrough=1



The code is not working i guess you missed adding the package 📦

**Comment by pratyushsingha** [comment]
https://github.com/Abbhiishek/codecommunitymusic-client/assets/111200225/1e559c50-255a-462e-94ba-3c4fb61db6a6

**Comment by pratyushsingha** [comment]
@Abbhiishek but it's working...is there anything that i missed??

**Comment by Abbhiishek** [comment]
The above is your deployed link for the changes you have made 


Can you check if your package.json file is fine ?

**Comment by Abbhiishek** [comment]
Okay let me check 😀

**Comment by Abbhiishek** [comment]
I was able to find the flaw.


The password strength progress bar is only working   if the user inputs a capital letter 

If the user only enters a small letter it is not shown.


Can we solve this issue ?

**Comment by pratyushsingha** [comment]
> I was able to find the flaw.
> 
> The password strength progress bar is only working if the user inputs a capital letter
> 
> If the user only enters a small letter it is not shown.
> 
> Can we solve this issue ?

okey i am trying to do so🚀

**Comment by pratyushsingha** [comment]
@Abbhiishek I fixed the bug

File: components/signup/signupform.tsx [hunk 1]
```diff
@@ -259,20 +279,74 @@ export default function SignUpForm() {
...
 
 
-    return (
-        <div className="flex flex-row justify-start gap-2">
-            <div
-                className={
-                    `h-2 w-full rounded-full ${passwordreset >= 1 ? "bg-green-500" : "bg-gray-300"}`
-                }
-            >
+//     return (
```
Comment (Abbhiishek) [inline components/signup/signupform.tsx:287]: @pratyushsingha can you please clean the lines which are comment out.

File: components/signup/signupform.tsx [hunk 2]
```diff
@@ -259,20 +279,74 @@ export default function SignUpForm() {
...
+
+    const changeColor = () => ({
+        width: `${barWidth}%`,
+        background: barColor(),
+        height: "7px",
+        borderRadius: "20px",
+        transition: "width 0.5s",
+    })
+    return (
+        <div className="bg-[#444654] rounded-xl h-2" >
```
Comment (Abbhiishek) [inline components/signup/signupform.tsx:348]: image


can you change the bg color to anyone shade from the above, Try to use tailwind class

**Review by Abbhiishek — CHANGES_REQUESTED** [review]
Everything looks great. Just clean the file a bit and change the strength bar background color with a tailwind class


Also, kindly add these contributions to your config/contributors file

File: components/signup/signupform.tsx (hunk 1 above)
Comment (pratyushsingha) [reply components/signup/signupform.tsx:287]: ok

File: config/contributors.ts [hunk 3]
```diff
@@ -6,11 +6,11 @@ export const Contributor: IContributor[] = [
     role: 'Frontend Developer',
     contributions: [
       {
-        title: 'add show/hide password feature for sign-in and login page',
```
Comment (Abbhiishek) [inline config/contributors.ts:9]: /create a new list item with the details .. leave the previous one as before :)

also am able to see that you haven't pulled the code. Kindly pull the code to resolve merge conflicts

**Review by Abbhiishek — CHANGES_REQUESTED** [review]

File: config/contributors.ts [hunk 4]
```diff
@@ -21,21 +22,20 @@ export const Contributor: IContributor[] = [
     name: 'Pratyush Singha',
     role: 'Frontend Developer',
```
Comment (Abbhiishek) [inline config/contributors.ts:23]: Hey @pratyushsingha  this seems to be a valid contribution that you made. Why you removed that from your list :)

**Review by Abbhiishek — CHANGES_REQUESTED** [review]

**Pull request merged**